public/app/ui/responsive.js

```diff
@@ -0,0 +1,30 @@
+(function () {
+    'use strict';
+
+    define(
+        [
+            'jquery'
+        ],
+        function ($) {
+            var responsive;
+
+            responsive = function () {
+                $(function () {
+                    $(window).on('resize', responsive.update);
+                    responsive.update();
+                });
+            };
+
+            responsive.update = function () {
```

Comment:
This is the new functionality.  It simply compares the sum of the heights of the `header.navbar` + `#application-container` + `footer.navbar` with the height of the `window` and sets the appropriate class on the footer.